backend/app/api/endpoints/media.py

```diff
@@ -0,0 +1,215 @@
+from fastapi.responses import FileResponse
+from pydantic import BaseModel
+
+from app.api.dependencies import is_valid_uuid
+from app.schemas.base import BaseIDNameModel
+
+router = APIRouter(prefix="/api/media", tags=["Media"])
+
+MEDIA_RESPONSE_LIST_EXAMPLES = {
+    "basic": Example(
+        summary="Paginated list of media items",
+        description="A sample paginated response containing media items with metadata",
+        value={
+            "data": [
+                {
+                    "id": "b0feaabc-da2b-442e-9b3e-55c11c2c2ff2",
+                    "name": "cat.jpg",
+                },
+                {
+                    "id": "c1feaabc-da2b-442e-9b3e-55c11c2c2ff3",
+                    "name": "dog.png",
+                },
+            ],
+            "pagination": {
+                "page": 1,
+                "page_size": 20,
+                "total_items": 2,
+                "total_pages": 1,
+            },
+        },
+    ),
+    "empty_response": Example(
+        summary="Empty result with pagination",
+        description="Response when no media items match the filter criteria",
+        value={
+            "data": [],
+            "pagination": {
+                "total_count": 0,
+                "page": 1,
+                "page_size": 20,
+                "total_pages": 0,
+            },
+        },
+    ),
+}
+
+
+def get_media_id(media_id: str) -> UUID:
+    """Initializes and validates a media ID"""
+    if not is_valid_uuid(media_id):
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Invalid media ID")
+    return UUID(media_id)
+
+
+# TODO: replace models with more specific variants from .schema package when defined
+class Media(BaseIDNameModel):
+    """Model representing media information"""
+
+    model_config = {
+        "json_schema_extra": {
+            "example": {
+                "id": "b0feaabc-da2b-442e-9b3e-55c11c2c2ff2",
+                "name": "cat.jpg",
+            },
+        }
+    }
+
+
+class PaginationMetadata(BaseModel):
+    """Pagination metadata for list responses"""
+
+    total_count: int
+    page: int
+    page_size: int
+    total_pages: int
+
+
+class MediaListResponse(BaseModel):
+    """Response model for paginated media list"""
+
+    data: list[Media]
+    pagination: PaginationMetadata
+
+
+@router.get(
+    "",
+    response_model=MediaListResponse,
+    responses={
+        status.HTTP_200_OK: {
+            "description": "List of available media",
+            "content": {
+                "application/json": {
+                    "examples": MEDIA_RESPONSE_LIST_EXAMPLES,
+                }
+            },
+        },
+        status.HTTP_400_BAD_REQUEST: {"description": "Invalid query parameters"},
+    },
+)
+async def list_media(
+    page: int = Query(default=1, ge=1, description="Page number (starts from 1)"),
+    page_size: int = Query(default=20, ge=1, le=100, description="Number of items per page (max 100)"),
```

Comment:
I think offset-based pagination would be a slightly better choice because:

1. It maps directly to SQLite `LIMIT ... OFFSET ...` clause
2. Unlike the page-based approach, it allows to retrieve any range even if not aligned with the page size (e.g. the range [10-29])